docs/AWSAdapter.org

```diff
@@ -13,37 +13,96 @@ The =AWSAdapter= is an adapter for the AWS platform.
 
 * Requirements
   1. =AWSAdapter= will create a VM on AWS based on the specification
-     given in the lab spec for a particular lab.
+     given in the lab spec for a particular lab. 
      
-  2. =AWSAdapter= will copy the =VMManager= component and start the
-     =VMManager= service on port 9089 on the newly created VM.
+  2. =AWSAdapter= will copy the =VMManager= component to the newly
+     created VM and start the =VMManager= service on port 9089.
 
 * Design
 
-** General Adapter Design 
+** General Adapter Design (REPHRASE) 
 According to ADS, every adapter must have certain interfaces. These
 interfaces enable other services to communicate to VMs, created by the
 adapter. The essential interfaces are listed below:
-1. create_vm(lab_spec)  TODO: Also capture signature of these functions
-2. start_vm
-3. stop_vm
-4. restart_vm
-5. start_vm_manager
-6. destroy_vm
-7. is_running_vm
 
+1. *create_vm* (self, lab_spec, dry_run=False)  
+      Creates a new VM. 
+      *Parameters*: lab_spec
+      *Return type*:
```

Comment:
Return type is not needed. As python implements dynamic typing, it is not possible to force return type apart from in conventions of coding.